File: php/streaming_pipe_in_and_out.php
<?php

$output_pipe = fopen('streaming-out.pipe', 'w');
$input_pipe = fopen('streaming-in.pipe', 'r');

while (($line = fgets($input_pipe)) !== false) {
    fwrite($output_pipe, "received " . $line);
}

fclose($input_pipe);
fclose($output_pipe);

File: swift/streaming_pipe_in_and_out.swift
import Foundation

/*
  Like readLine(), but for reading from a file/pipe
*/
func readFileLine(file: UnsafeMutablePointer<FILE>) -> String? {
  var line: UnsafeMutablePointer<CChar>? = nil
  var n = 0
  defer { free(line) }
  return getline(&line, &n, file) > 0 ? String(cString: line!) : nil
}

let pipeIn = fopen("streaming-in.pipe", "r")!
let pipeOutHandle = open("streaming-out.pipe", O_WRONLY)

while let line = readFileLine(file: pipeIn) {
  let outputLine = "received \(line)"
  write(pipeOutHandle, outputLine, outputLine.utf8.count)
}

fclose(pipeIn)
